proof/invariant-abstract/LevityCatch_AI.thy

```diff
@@ -39,10 +39,12 @@ lemma const_on_failure_wp :
 
 lemma get_cap_id:
   "(v, s') \<in> fst (get_cap p s) \<Longrightarrow> (s' = s)"
-  by (clarsimp simp: get_cap_def get_object_def in_monad
-                     split_def
-              split: Structures_A.kernel_object.splits)
-
+  by (clarsimp simp: get_cap_def get_object_def read_object_def gets_the_def
+                     split_def gets_def get_def assert_opt_def bind_def
+                     return_def fail_def assert_def
+              split: option.splits)
+     (rename_tac ko a; case_tac ko;
```

Comment:
Just a thought: I've seen quite a few cases where we have to make up variable names if we'd like to `rename_tac` something that's not last in the list of green variables. I guess we'd normally want to avoid a situation where we're `rename_tac`ing the, say, 9th of 16 new green variables, but it would be cool if we could use `_` here (I'm pretty sure that doesn't work currently)

Reply:
`rename_tac` doesn't accept `_` arguments, yes. If someone wants to dig into it, that could probably be changed by treating `_` specially and only doing the underlying rename for the remaining arguments. The `subgoal` command does support the `_` syntax.  The change would have to be made upstream, otherwise we'd be incompatible with standard Isabelle.